hub/util/remove_cache.py

```diff
@@ -41,3 +42,21 @@ def get_dataset_with_zero_size_cache(ds):
     if ds.pending_commit_id != commit_id:
         ds.checkout(commit_id)
     return ds
+
+
+def create_read_copy_dataset(dataset, commit_id=None):
```

Comment:
Please add few liner docstring to understand why this function exists and what it actually does. 

Does it copy the data or creates a new "dataset" with read only access to chunks?

Reply:
Looks like second, right?

Reply:
added docstring